src/audio/codec_adapter/codec/cadence.c

```diff
@@ -381,9 +415,21 @@ int cadence_codec_prepare(struct comp_dev *dev)
 	int ret = 0, mem_tabs_size;
 	struct codec_data *codec = comp_get_codec(dev);
 	struct cadence_codec_data *cd = codec->private;
+	uint32_t api_id = CODEC_GET_API_ID(codec->id);
+
+	/* req_api_id will be passed via parameters from the linux
+	 * kernel and it will contain the api_id of the file format.
+	 */
+	uint32_t req_api_id = CADENCE_CODEC_MP3_DEC_ID;
```

Comment:
Is this some sort of default?

Reply:
As mentioned in the commit message, we use this as default for now because codec id should be sent from the kernel side. Why don't have a clean way to do that now.